source/Halibut/Transport/SecureListener.cs

```diff
@@ -277,17 +294,21 @@ void SendFriendlyHtmlPage(Stream stream)
             }
         }
 
-        bool Authorize(SslStream stream, EndPoint clientName)
+        static string GetThumbprint(SslStream stream)
         {
-            log.Write(EventType.Diagnostic, "Begin authorization");
-
             if (stream.RemoteCertificate == null)
             {
-                log.Write(EventType.ClientDenied, "A client at {0} connected, and attempted a message exchange, but did not present a client certificate", clientName);
-                return false;
+                return null;
             }
 
             var thumbprint = new X509Certificate2(stream.RemoteCertificate.Export(X509ContentType.Cert)).Thumbprint;
+            return thumbprint;
+        }
+
+        bool Authorize(string thumbprint, EndPoint clientName)
```

Comment:
Consider putting a guard on the thumbprint != null, now that the null check has been moved out of here.